events-api/src/modules/tickets/tickets.service.ts

```diff
@@ -0,0 +1,144 @@
+import { BadRequestException, HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { DatabaseService } from '../database/database.service';
+import { CreateTicketDto } from './dto/create-ticket-dto';
+import { CreatePaymentDto } from './dto/create-payment.dto';
+import { lastValueFrom } from 'rxjs';
+import { HttpService } from '@nestjs/axios';
+
+@Injectable()
+export class TicketsService {
+  constructor(
+    private readonly databaseService: DatabaseService,
+    private readonly httpService: HttpService,
+  ) {}
+
+  async createTicket(createTicketDto: CreateTicketDto, userId: string): Promise<CreateTicketDto> {
+    try {
+      if (createTicketDto.ticketQuantity <= 0) {
```

Comment:
Validation should be mostly done with DTO, in this case, you can do such validation on the DTO level